tests/unit/lms/services/digest/service_test.py

```diff
@@ -0,0 +1,150 @@
+from unittest.mock import create_autospec, patch, sentinel
+
+import pytest
+from h_matchers import Any
+
+from lms.services.digest._digest_assistant import DigestAssistant
+from lms.services.digest._models import Digest, HCourse, HUser
+from lms.services.digest.service import DigestService
+from lms.services.mailchimp import EmailRecipient, EmailSender
+
+
+class TestDigestService:
+    def test_send_emails(self, svc, get_digests, mailchimp_service, digest, sender):
+        svc.send_emails(sentinel.audience, sentinel.after, sentinel.before)
+
+        get_digests.assert_called_once_with(
+            sentinel.audience, sentinel.after, sentinel.before
+        )
+        mailchimp_service.send_template.assert_called_once_with(
+            template_name="instructor-email-digest",
+            sender=sender,
+            recipient=EmailRecipient(email="email@1", name="Display Name"),
+            template_vars=digest.serialize(),
+        )
+
+    @pytest.mark.usefixtures("get_digests")
+    @pytest.mark.parametrize(
+        "user_email,override_to_email,send_email",
+        (
+            ("email@example.com", None, "email@example.com"),
+            (None, "OVERRIDE@example.com", "OVERRIDE@example.com"),
+            ("email@example.com", "OVERRIDE@example.com", "OVERRIDE@example.com"),
+            (None, None, None),
+        ),
+    )
+    def test_send_email_destination(
+        self, svc, digest, mailchimp_service, user_email, override_to_email, send_email
+    ):
+        digest.audience_user.email = user_email
+
+        svc.send_emails(
+            sentinel.audience,
+            sentinel.after,
+            sentinel.before,
+            override_to_email=override_to_email,
+        )
+
+        if send_email:
+            mailchimp_service.send_template.assert_called_once_with(
+                template_name=Any(),
+                sender=Any(),
+                recipient=EmailRecipient(email=send_email, name=Any()),
+                template_vars=Any(),
+            )
+        else:
+            mailchimp_service.send_template.assert_not_called()
+
+    def test_send_email_requires_some_annotations(self, svc, digest, mailchimp_service):
+        digest.courses[0].annotations = []
+
+        svc.send_emails(sentinel.audience, sentinel.after, sentinel.before)
+
+        mailchimp_service.send_template.assert_not_called()
+
+    def test_get_digests(self, svc, h_api, digest_assistant):
```

Comment:
We aren't testing lots of different combinations of things, but I'm not sure how many it's worth testing. A lot of the edge casey things we were testing here before are now someone else's responsiblity.

As long as the annotations get on the right course and assigned to the user it's all gravy.

The two ways to go with this I think are:

 * Add different tests which cover different aspects, but leaves us with a horrible fixturing problem
 * Add some more elements to this test to cover more cases